packages/loot-core/esbuild.config.mjs

```diff
@@ -0,0 +1,81 @@
+import fs from 'fs';
+import * as path from 'path';
+
+import * as esbuild from 'esbuild';
+import { polyfillNode } from 'esbuild-plugin-polyfill-node';
+import peg from 'peggy';
+
+/// TODO: this is super hacky.
+// import { createRequire } from 'node:module';
+// const require = createRequire(import.meta.url);
+
+const pegPlugin = {
+  name: 'peg',
+  setup: build => {
+    build.onResolve({ filter: /\.pegjs$/ }, args => ({
+      path: path.isAbsolute(args.path)
+        ? args.path
+        : path.join(args.resolveDir, args.path),
+      namespace: 'peg',
+    }));
+
+    build.onLoad({ filter: /.*/, namespace: 'peg' }, async args => {
+      let text = await fs.promises.readFile(args.path, 'utf8');
+      return {
+        loader: 'js',
+        // TODO: investigate options
+        //       https://peggyjs.org/documentation#generating-a-parser-javascript-api
+        //       Output can include a source map, which may be useful!
+        contents: peg.generate(text, { output: 'source', format: 'umd' }),
+      };
+    });
+  },
+};
+
+const IS_DEV = process.env.NODE_ENV === 'development';
+
+const config = {
+  entryPoints: [{ out: 'kcab.worker', in: 'src/server/main.js' }],
+  bundle: true,
+  outdir: 'lib-dist/browser',
+  globalName: 'backend',
+  publicPath: '/kcab/',
+  minify: !IS_DEV,
+  sourcemap: IS_DEV,
+  entryNames: process.env.OUTFMT ?? '[dir]/[name]',
+  loader: {
+    '.ts': 'ts',
+    '.web.ts': 'ts',
+    '.js': 'js',
+    '.web.js': 'js',
+  },
+  resolveExtensions: [
+    '.web.js',
+    '.web.ts',
+    '.web.tsx',
+    '.js',
+    '.ts',
+    '.tsx',
+    '.json',
+  ],
+  define: {
+    'process.env.IS_DEV': JSON.stringify(IS_DEV),
+    'process.env.IS_BETA': JSON.stringify(
+      process.env.ACTUAL_RELEASE_TYPE === 'beta',
+    ),
+    'process.env.PUBLIC_URL': JSON.stringify(process.env.PUBLIC_URL || '/'),
+    'process.env.ACTUAL_DATA_DIR': JSON.stringify('/'),
+    'process.env.ACTUAL_DOCUMENT_DIR': JSON.stringify('/documents'),
+  },
+  alias: {
+    // fs: require.resolve('memfs'),
```

Comment:
Everything seems to be working without aliasing `fs`, and `polyfillNode` does not polyfill `fs` by default. Is there a good way to test that everything still works as intended? `yarn test:web` is passing, and my local tests seem to work.